plothist/comparison.py

```diff
@@ -0,0 +1,341 @@
+
+    """
+    for h in hist_list:
+        if not len(h.axes) == len(hist_list[0].axes):
+            raise ValueError("Histograms must have same dimensionality.")
+        for i in range(len(h.axes)):
+            if not h.axes[i] == hist_list[0].axes[i]:
+                raise ValueError("The bins of the histograms must be equal.")
+
+
+def _hist_ratio_variances(hist_1, hist_2):
+    """
+    Calculate the variances of the ratio of two histograms (hist_1/hist_2).
+
+    Parameters
+    ----------
+    hist_1 : boost_histogram.Histogram
+        The first histogram.
+    hist_2 : boost_histogram.Histogram
+        The second histogram.
+
+    Returns
+    -------
+    variances : np.ndarray
+        The variances of the ratio of the two histograms.
+
+    Raises
+    ------
+    ValueError
+        If the bins of the histograms are not equal.
+    """
+
+    _check_binning_consistency([hist_1, hist_2])
+
+    np.seterr(divide="ignore", invalid="ignore")
+    ratio_variances = np.where(
+        hist_2.values() != 0,
+        hist_1.variances() / hist_2.values() ** 2
+        + hist_2.variances() * hist_1.values() ** 2 / hist_2.values() ** 4,
+        np.nan,
+    )
+    np.seterr(divide="warn", invalid="warn")
+
+    return ratio_variances
+
+
+def compute_pull(hist_1, hist_2, poisson_for_hist_1=False):
+    """
+    Compute the pull between two histograms.
+
+    Parameters
+    ----------
+    hist_1 : boost_histogram.Histogram
+        The first histogram.
+    hist_2 : boost_histogram.Histogram
+        The second histogram.
+    poisson_for_hist_1 : bool, optional
+        Whether to use Poisson uncertainties for hist_1. Default is False.
+
+    Returns
+    -------
+    comparison_values : numpy.ndarray
+        The pull values.
+    comparison_uncertainties_low : numpy.ndarray
+        The lower uncertainties on the pull. Always ones.
+    comparison_uncertainties_high : numpy.ndarray
+        The upper uncertainties on the pull. Always ones.
+    """
+    if poisson_for_hist_1:
+        uncertainties_low, uncertainties_high = _get_poisson_uncertainties(hist_1)
+        hist_1_variances = np.where(
+            hist_1.values() >= hist_2.values(),
+            uncertainties_low**2,
+            uncertainties_high**2,
+        )
+        hist_1 = hist_1.copy()
+        hist_1[:] = np.c_[hist_1.values(), hist_1_variances]
+
+    comparison_values = np.where(
+        hist_1.variances() + hist_2.variances() != 0,
+        (hist_1.values() - hist_2.values())
+        / np.sqrt(hist_1.variances() + hist_2.variances()),
+        np.nan,
+    )
+    comparison_uncertainties_low = np.ones_like(comparison_values)
+    comparison_uncertainties_high = comparison_uncertainties_low
+
+    return (
+        comparison_values,
+        comparison_uncertainties_low,
+        comparison_uncertainties_high,
+    )
+
+
+def compute_difference(hist_1, hist_2, poisson_for_hist_1=False):
```

Comment:
Same comment as for `pull`. A bit strange that the function is not symmetrical for h1 and h2. Why this poisson_for_hist_1 argument? why not also a poisson_for_hist_2? We could use kwargs.